src/swarm/swarm-tools.ts

```diff
@@ -0,0 +1,98 @@
+import { z } from 'zod';
+
+import {
+  extractZodObject,
+  type Msg,
+  MsgUtil,
+  zodToJsonSchema,
+} from '../model/index.js';
+import { getErrorMsg } from '../model/utils/errors.js';
+import { cleanString } from '../model/utils/message-util.js';
+import { type Agent, type SwarmFunc } from './types.js';
+
+export function swarmFunction<Schema extends z.ZodObject<any>, Return>(
```

Comment:
curious why this is necessary? seems the same as `createAIFunction` (and prolly should be the same, no?)

Reply:
i'm honestly not sure. i plan to almost entirely rewrite both of these though. they were just copy pasta from ai function, as you noticed.